pkg/controller/common/webhook/webhook_test.go

```diff
@@ -0,0 +1,227 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	admissionv1 "k8s.io/api/admission/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	agentv1alpha1 "github.com/elastic/cloud-on-k8s/pkg/apis/agent/v1alpha1"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/k8s"
+	"github.com/elastic/cloud-on-k8s/pkg/utils/set"
+)
+
+func asJSON(obj interface{}) []byte {
+	data, err := json.Marshal(obj)
+	if err != nil {
+		panic(err)
+	}
+	return data
+}
+
+func Test_validatingWebhook_Handle(t *testing.T) {
+	type fields struct {
+		managedNamespaces set.StringSet
+		validator         admission.Validator
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		req    admission.Request
+		want   admission.Response
+	}{
+		{
+			"create properly validates valid agent, and returns allowed.",
```

Comment:
I would be in favour of using keys in these structs instead of positional initialisation.

Reply:
No problem, and done.  I'll merge this after ensuring all of the checks pass again.